File: TwoSum.py
# https://leetcode.com/problems/two-sum/
# Given an array of integers, return indices of the two numbers such that they add up to a specific target.

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        compliment_map = dict()
        
        for i in range(len(nums)):
            compliment  = target - nums[i]

            if nums[i] in compliment_map.keys():
                return [compliment_map[nums[i]],i]
            compliment_map.setdefault(compliment,i)
            

File: twosum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int[] result = new int[2];
        Dictionary<int,int> compliment = new Dictionary<int,int>();
        for(int i = 0; i <nums.Length; i++){
            int value = target - nums[i];
            compliment.TryAdd(value,i);
        }
        for (int i= 0; i < nums.Length; i++){
            if (compliment.ContainsKey(nums[i])){
                if (i == compliment[nums[i]])
                    continue;
                result[0] = i;
                result[1] = compliment[nums[i]];
                return result;
            }
                
        }
        return result;
    }
}
